keps/sig-api-machinery/20190603-immutable-fields.md

```diff
@@ -420,34 +503,44 @@ immutability of fields.
 
 - exhaustive unit tests are added in apiextensions-apiserver for
   - CRD validation
-    - for `x-kubernetes-immutable: true` and `x-kubernetes-immutable-keys: true` at the root and in `metadata`
-    - for `x-kubernetes-immutable-keys: true` for list-type `map` and forbidden otherwise
-    - for `x-kubernetes-immutable: true` on the list-keys of `x-kubernetes-immutable-keys: true` maps.
-    - for `x-kubernetes-immutable: true` and `x-kubernetes-immutable-keys: true` only for v1 CRDs, or during ratcheting updates.
-  - immutability checking with all variants of `x-kubernetes-immutable`, `x-kubernetes-immutable-keys` and `x-kubernetes-list-type`.
+    - for `x-kubernetes-mutability` and `x-kubernetes-key-mutability` at the root and in `metadata`
+    - for `x-kubernetes-key-mutability` forbidden for object with `properties`
+    - for `x-kubernetes-mutability: AddOnly | RemoveOnly` forbidden for objects without `properties` (maps) and arrays.
```

Comment:
this is a bit ambious, 
```
foo:
      type: string
      x-kubernetes-mutability: Immutable
```
is a valid object with `x-kubernetes-mutability: Immutable` and without properties. We might want to specify the compatible "type: *" here

Reply:
It's not an object.

Reply:
clarified